### PR TITLE
fix(storage): replace BufferSource with subtypes for types to get picked up

### PR DESCRIPTION
**Pull request by kvramyasri7**
#### Description of changes

- replace BufferSource with subtypes for types to get picked up
Before

image
image

After

image
image

- remove file type from `StorageUploadDataPayload`. This change is missed  sent previous PR to `next/release`

#### Description of how you validated changes

Build library with verdaccio, install in RN sample app and see the types

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/src/types/inputs.ts [hunk 1]
```diff
@@ -49,4 +49,4 @@ export type StorageCopyInput<
 /**
  * The data payload type for upload operation.
  */
-export type StorageUploadDataPayload = Blob | BufferSource | string | File;
+export type StorageUploadDataPayload = Blob | BufferSource | string;
```
Comment (AllanZhengYP) [inline packages/storage/src/types/inputs.ts:52]: I think I found a more organic solution to make `File` type available according to the runtime:

```suggestion
export type StorageUploadDataPayload =
	| Blob
	| ArrayBufferView
	| ArrayBuffer
	| string
	| OptionalFile;

/**
 * File type is not available in environments like React Native. This type
 * returns File type only when it's defined in global.
 */
type OptionalFile = unknown extends File ? never : File;
```

The `BufferSource` is replaced by `ArrayBufferView | ArrayBuffer` explicitly and they are equivalent.

Comment (AllanZhengYP) [reply packages/storage/src/types/inputs.ts:52]: In browser:
Screenshot 2023-10-12 at 3 28 11 PM
In RN:
Screenshot 2023-10-12 at 3 28 19 PM

Comment (kvramyasri7) [reply packages/storage/src/types/inputs.ts:52]: I think for removing `File` we are creating new api this change is to support `uploadFile` api. What do you think ?

Comment (kvramyasri7) [reply packages/storage/src/types/inputs.ts:52]: Thanks for the suggestion I will change the types and test in sample app.

**Review by AllanZhengYP — APPROVED** [review]
It looks good to me!

**Review by israx — APPROVED** [review]
lgtm

**Review by HuiSF — APPROVED** [review]

**Pull request merged**